knative-operator/deploy/resources/kourier/download.sh

```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-KOURIER_VERSION=release-0.15
+KOURIER_VERSION=release-0.16
```

Comment:
Ideally. We move this to hack/lib/vars.bash to have all these versions in one place. But  that can be done later.